File: test_data/code/java/1796-second-largest-digit-in-a-string.java
class Solution {
    public int secondHighest(String s) {
        Set<Integer> set = new HashSet<>();
        for (char c : s.toCharArray()) {
            if (Character.isDigit(c)) {
                set.add(Integer.parseInt(String.valueOf(c)));
            }
        }
        if (set.size() < 2) {
            return -1;
        }
        int secondHighest = -1;
        for (int num : set) {
            if (num > secondHighest && num != Collections.max(set)) {
                secondHighest = num;
            }
        }
        return secondHighest;
    }
}

File: test_data/code/python/1796-second-largest-digit-in-a-string.py
class Solution:
    def secondHighest(self, s: str) -> int:
        digits = set()
        for c in s:
            if c.isdigit():
                digits.add(int(c))
        digits = list(sorted(digits, reverse=True))
        if len(digits) < 2:
            return -1
        return digits[1]